common/flatpak-dir.c

```diff
@@ -12718,6 +12737,20 @@ flatpak_dir_modify_remote (FlatpakDir   *self,
                imported, (imported == 1) ? "" : "s", remote_name);
     }
 
+  filter_path = g_key_file_get_value (new_config, group, "xa.filter", NULL);
+  if (filter_path && *filter_path && g_file_test (filter_path, G_FILE_TEST_EXISTS))
+    {
+      /* Make a backup filter copy in case it goes away later */
+      char *filter_name = g_strconcat (remote_name, ".filter", NULL);
+      g_autoptr(GFile) filter_file = g_file_new_for_path (filter_path);
+      g_autoptr(GFile) filter_copy = flatpak_build_file (self->basedir, "repo", filter_name, NULL);
+      g_autoptr(GError) local_error = NULL;
+
+      if (!g_file_copy (filter_file, filter_copy, G_FILE_COPY_OVERWRITE | G_FILE_COPY_TARGET_DEFAULT_PERMS,
+                        cancellable, NULL, NULL, &local_error))
```

Comment:
Convenient to just g_file_copy, but I think it would be useful to add a comment that says:

```
# backup copy of ORIG_PATH, do not edit!
```